recipe/build.sh

```diff
@@ -19,7 +19,6 @@ cmake \
     -D "CMAKE_INSTALL_PREFIX=${PREFIX}" \
     -D "CMAKE_PROGRAM_PATH=${BUILD_PREFIX}" \
```

Comment:
The default value on conda-forge is `-DCMAKE_PROGRAM_PATH="$BUILD_PREFIX/bin;$PREFIX/bin" \
`, which is part of the `CMAKE_ARGS` env var set during compiler activation. Ideally, we should deviate as little as possible from the default configurations. A regular CMake incantation looks like:

```bash
cmake ${CMAKE_ARGS} ${PROJECT_FLAGS} ${SRC_DIR}
```

where `PROJECT_FLAGS` contains the project-specific definitions like `-D SimpleITK_BUILD_DISTRIBUTE:BOOL=ON` and friends.

I wonder if we can make it look like what we have now in https://github.com/conda-forge/simpleitk-feedstock/pull/27. If that's not possible, I like having comments next to the lines that "don't follow the rule" to make it easier for contributors in the future. That's my personal preference, at least.

Reply:
On OSX I see: 
```
CMAKE_ARGS=-DCMAKE_AR=$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-ar -DCMAKE_CXX_COMPILER_AR=$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-ar -DCMAKE_C_COMPILER_AR=$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-ar -DCMAKE_RANLIB=$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-ranlib -DCMAKE_CXX_COMPILER_RANLIB=$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-ranlib -DCMAKE_C_COMPILER_RANLIB=$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-ranlib -DCMAKE_LINKER=$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-ld -DCMAKE_STRIP=$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-strip -DCMAKE_INSTALL_NAME_TOOL=$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-install_name_tool -DCMAKE_LIBTOOL=$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-libtool -DCMAKE_OSX_DEPLOYMENT_TARGET=10.9 -DCMAKE_BUILD_TYPE=Release -DCMAKE_OSX_SYSROOT=/Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk -DCMAKE_INSTALL_PREFIX=$PREFIX -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_PROGRAM_PATH=$BUILD_PREFIX/bin;$PREFIX/bin
```
Which does not include the `CMAKE_FIND_ROOT_*` options that are currently coded here or that are on linux.

Reply:
Looking over the cmake args here, it looks like OS X deployment target and sys root would be missing from the configuration for libsimpleitk, but it would be in libitk and simpleitk.